### PR TITLE
Music: embeddable mini-player

### PR DESCRIPTION
**Pull request by breville**
This explores an embeddable version of the Music Lab mini-player that can be used on third-party sites.  

Designed for internal use for now, it will let us test whether the cross-domain requests work.

It can be embedded in an `iframe` with a comma-separated list of channel IDs, such as https://studio.code.org/musiclab/embed?channels=channelID1,channelID2,channelID3

### CODE REVIEW

File: dashboard/app/controllers/musiclab_controller.rb [hunk 1]
```diff
@@ -55,6 +55,26 @@ def gallery
       to_json
   end
 
+  def embed
+    response.headers['X-Frame-Options'] = 'ALLOWALL'
+    response.headers['Content-Security-Policy'] = ''
```
Comment (breville) [inline dashboard/app/controllers/musiclab_controller.rb:60]: Borrowed from: https://github.com/code-dot-org/code-dot-org/blob/db70996804531e81bb83452572b9f1aaca72ac98/dashboard/app/controllers/projects_controller.rb#L372-L375

File: dashboard/config/routes.rb [hunk 2]
```diff
@@ -45,6 +45,7 @@
     get "/projectbeats", to: "musiclab#index"
     get "/musiclab/menu", to: "musiclab#menu"
     get "/musiclab/gallery", to: "musiclab#gallery"
+    get "/musiclab/embed", to: "musiclab#embed"
```
Comment (sanchitmalhotra126) [inline dashboard/config/routes.rb:48]: minor: what do you think about calling this `musiclab/playlist_embed` / `embed_playlist` or similar?

edit: or alternatively `/playlist/embed` if we think that we might use `/playlist` to display the same content just on our site (without an iframe)?

**Review by sanchitmalhotra126 — APPROVED** [review]
LGTM 👍

File: dashboard/config/routes.rb (hunk 2 above)
Comment (breville) [reply dashboard/config/routes.rb:48]: Good question.  Given that we are pausing this feature work right now, beyond this cross-site testing, I think I'll hold off on a rename until we learn more about what happens next.  But agreed that differentiating this from other types of embeds is probably a good next move.

**Pull request merged**